c10/util/env.h

```diff
@@ -0,0 +1,74 @@
+#pragma once
+
+#include <c10/util/Exception.h>
+#include <cstring>
+#include <iostream>
+#include <sstream>
+
+namespace c10 {
+namespace utils {
+// Reads an environment variable and returns
+// - true,              if set equal to "1"
+// - false,             if set equal to "0"
+// - `default_value`,   otherwise
+//
+// NB:
+// Issues a warning if the value of the environment variable is not 0 or 1.
+bool check_env_bool(const char* name, bool default_value = false) {
+  auto envar = std::getenv(name);
+  if (envar) {
+    if (strcmp(envar, "0") == 0) {
+      return false;
+    }
+    if (strcmp(envar, "1") == 0) {
+      return true;
+    }
+    TORCH_WARN(
+        "Ignoring invalid value for boolean flag ",
+        name,
+        ": ",
+        envar,
+        "valid values are 0 or 1.");
+  }
+  return default_value;
+}
+
+// Reads an environment variable and returns
+// - its value,         if it is set and is a valid value
+// - `default_value`,   otherwise
+//
+// You can optionally pass in a list of valid values (default: empty list,
+// which is interpreted as "all values accepted")
+std::string check_env(
```

Comment:
What are some of the uses for this? If it's just `PYTORCH_JIT_LOG_LEVEL` I think we can probably just get rid of it since it'd end up being pretty complicated and there's not many other places it'd be used in the codebase (so the extra code complexity wouldn't be worth it IMO)

Reply:
- `torch/lib/c10d/ProcessGroupNCCL.cpp` (default value of `UNSET`, allowed values: `*`)
- `torch/lib/c10d/Utils.cpp` (parse_env utility, allowed values `[TORCH_DISTRIBUTED_DEBUG, DETAIL, INFO, OFF]`

These two files made me want to add this utility. I took a look at the rest of the `getenv` uses, and they either fit in to the boolean category, or they have their own parsing logic in place. I can take this out.